crates/wasm-pkg-core/src/config.rs

```diff
@@ -75,56 +74,22 @@ pub struct Override {
 pub struct Metadata {
     /// The authors of the package.
     #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub authors: Option<Vec<String>>,
```

Comment:
I think for one version we should probably keep these, aliasing where it makes sense (e.g. mapping the first item in the `authors` vec to `author`, aliasing the json deserialization for `licenses`) and deprecating the other fields. The latter might not matter as much because the deserializer should just ignore unknown fields.

Reply:
I'm on the fence here due to a couple of issues:

- There is no equivalent for `documentation` or `categories` so these would be silently discarded
- The URLs are validated as...URLs
- The new wasm-metadata validates license SPDX formatting

Given these basically unavoidable incompatibilities it feels like we might as well rip the band-aid off on this one unless there are existing consumers we know would benefit from a transition.

> the deserializer should just ignore unknown fields.

I wonder if we should change this; accepting e.g. misspelled fields is kind of gnarly UX... :grimacing:

Reply:
I added aliases for `license` and `repository` as those seem harmless; skipped `authors` since the field type changed and there is no canonical OCI way to specify multiple authors. Added `deny_unknown_fields`.